tests/unit/charms/lte_core_interface/v0/test_lte_core_interface_provider.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import unittest
+from ipaddress import AddressValueError
+
+import pytest
+from ops import testing
+
+from tests.unit.charms.lte_core_interface.v0.dummy_provider_charm.src.charm import (
+    DummyCoreProviderCharm,
+)
+
+testing.SIMULATE_CAN_CONNECT = True
+
+BASE_CHARM_DIR = "tests.unit.charms.lte_core_interface.v0.dummy_requirer_charm.src.charm.DummyCoreRequirerCharm"  # noqa: E501
+
+
+class TestCoreProvider(unittest.TestCase):
+    def setUp(self):
+        self.relation_name = "lte-core"
+        self.remote_app = "lte-core-requirer"
+        self.harness = testing.Harness(DummyCoreProviderCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    def test_given_unit_is_leader_and_remote_unit_joined_relation_when_set_mme_address_then_data_is_added_to_application_databag(  # noqa: E501
+        self,
+    ):
+        self.harness.set_leader(is_leader=True)
+        relation_id = self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=self.remote_app
+        )
+        mme_ipv4_address = "0.0.0.0"
+
+        self.harness.update_relation_data(
+            relation_id=relation_id,
+            app_or_unit=self.remote_app,
+            key_values={"mme_ipv4_address": mme_ipv4_address},
+        )
```

Comment:
Why do we need to update relation data manually? Test charm should take care of that as a result of `relation_changed`, shouldn't it?

Reply:
Agree, fixed.